src/xcode/ENA/ENA/Source/HTTPClient/Resources/_Resource.swift

```diff
@@ -41,17 +41,64 @@ extension Resource {
 }
 
 /**
-The errors that can occur while handling resources
-*/
-enum ResourceError: Error {
+ The errors that can occur while handling resources
+ */
+enum ResourceError: Error, Equatable {
 	case missingData
-	case decoding
+	case decoding(Error?)
 	case encoding
 	case packageCreation
 	case signatureVerification
 	case notModified
 	case undefined
 	case missingEtag
+	
+	// MARK: - Protocol Equatable
+	
+	// swiftlint:disable cyclomatic_complexity
+	static func == (lhs: ResourceError, rhs: ResourceError) -> Bool {
+		switch (lhs, rhs) {
+		case (.missingData, .missingData):
+			return true
+		case (.missingData, _):
+			return false
+			
+		case let (.decoding(lError), .decoding(rError)):
+			return lError?.localizedDescription == rError?.localizedDescription
+		case (.decoding, _):
+			return false
+			
+		case (.encoding, .encoding):
+			return true
+		case (.encoding, _):
+			return false
+			
+		case (.packageCreation, .packageCreation):
+			return true
+		case (.packageCreation, _):
+			return false
+			
+		case (.signatureVerification, .signatureVerification):
+			return true
+		case (.signatureVerification, _):
+			return false
+			
+		case (.notModified, .notModified):
+			return true
+		case (.notModified, _):
+			return false
+			
+		case (.undefined, .undefined):
+			return true
+		case (.undefined, _):
+			return false
+			
+		case (.missingEtag, .missingEtag):
+			return true
+		case (.missingEtag, _):
+			return false
+		}
+	}
```

Comment:
```suggestion
```

The manual implementation for enums with associated values should not be needed anymore since Swift 4.1, I think you can just remove this

Reply:
Yes, but only if i add the optionality and `.localizedDescription` in the `Equatable` implementation of `ServiceError`.

Reply:
But makes the code less complicated, so I can change it.